nautobot/utilities/tests/test_filters.py

```diff
@@ -769,3 +770,89 @@ def test_device_comments_multiple_value_charfield_iregex(self):
     def test_device_comments_multiple_value_charfield_iregex_negation(self):
         params = {"comments__nire": ["^device"]}
         self.assertEqual(self.DeviceFilterSetWithComments(params, self.device_queryset).qs.count(), 0)
+
+
+class SearchFilterTest(TestCase):
+    """Tests for the `SearchFilter` filter class."""
+
+    filterset_class = SiteFilterSet
+
+    def setUp(self):
+
+        super().setUp()
+
+        self.region1 = Region.objects.create(name="Test Region 1", slug="test-region-1")
+        self.region2 = Region.objects.create(name="Test Region 2", slug="test-region-2")
+        self.site1 = Site.objects.create(region=self.region1, name="Test Site 1", slug="test-site1", asn=1234)
+        self.site2 = Site.objects.create(region=self.region2, name="Test Site 2", slug="test-site2", asn=12345)
+        self.site3 = Site.objects.create(region=None, name="Test Site 3", slug="test-site3")
+
+        self.queryset = Site.objects.all()
+
+    def get_filterset_count(self, params, klass=None):
+        """To save ourselves some boilerplate."""
+        if klass is None:
+            klass = self.filterset_class
+        return klass(params, self.queryset).qs.count()
+
+    def test_default_icontains(self):
+        """Test a default search for an "icontains" value."""
+        params = {"q": "Test Site"}
+        self.assertEqual(self.get_filterset_count(params), 3)
+        params = {"q": "Test Site 3"}
+        self.assertEqual(self.get_filterset_count(params), 1)
+        # Trailing space should also match all
```

Comment:
Should we have a site that doesn't match the trailing space (`Test Site4` or whatever) in order to confirm that the query string is/isn't stripped?

Reply:
I am thus far unable to confirm this and not because of our code. It appears it's being stripped somewhere else upstream. Trying to chase it down.